project-base/storefront/helpers/gtm/gtm.ts

```diff
@@ -42,37 +42,33 @@ import {
 } from 'types/gtm/objects';
 
 export const useGtmCartInfo = (): { gtmCartInfo: GtmCartInfoType | null; isCartLoaded: boolean } => {
-    const { cart, isInitiallyLoaded, promoCode } = useCurrentCart();
+    const { cart, promoCode, isFetching } = useCurrentCart();
     const cartUuid = usePersistStore((store) => store.cartUuid);
     const { isUserLoggedIn } = useCurrentUserData();
     const domainConfig = useDomainConfig();
 
     return useMemo(() => {
-        if ((cartUuid === null && !isUserLoggedIn) || cart === null) {
-            return { gtmCartInfo: null, isCartLoaded: isInitiallyLoaded };
+        if ((!cartUuid && !isUserLoggedIn) || !cart) {
+            return { gtmCartInfo: null, isCartLoaded: !isFetching };
         }
 
         return {
             gtmCartInfo: getGtmMappedCart(cart, promoCode, isUserLoggedIn, domainConfig, cartUuid),
-            isCartLoaded: isInitiallyLoaded,
+            isCartLoaded: !isFetching,
         };
-    }, [cart, cartUuid, domainConfig, isInitiallyLoaded, isUserLoggedIn, promoCode]);
+    }, [cart, cartUuid, domainConfig, isFetching, isUserLoggedIn, promoCode]);
 };
 
 export const getGtmMappedCart = (
     cart: CartFragmentApi,
-    promoCode: string | null,
+    promoCode: string | null | undefined,
```

Comment:
I don't see why this is necessary :thinking:

Reply:
Seems like some leftover after my refactoring. Reverted.